### ISSUE TITLE
How do you reliably record state changes of the job progresses through the microservices?

### ISSUE THREAD

**Issue by vikaspro90** [airavata-courses/TeamSangam#96]

**Comment by vikaspro90** [comment]
We are already recording the state changes in the database. After each microservice completes execution, the outcome is written to the DB. I guess this should be enough to get the state of the job whenever required. What do you think?

**Comment by rajasimha1991** [comment]
Yes, as the database is getting updated with the requests and responses of each microservice, I think this will help in tracking the state changes of the job.
